src/lib/operators/join_hash.cpp

```diff
@@ -50,10 +53,13 @@ std::shared_ptr<const Table> JoinHash::_on_execute() {
   // This is the expected implementation for swapping tables:
   // (1) if left or right outer join, outer relation becomes probe relation (we have to swap only for left outer)
   // (2) for a semi and anti join the inputs are always swapped
-  bool inputs_swapped = (_mode == JoinMode::Left || _mode == JoinMode::Anti || _mode == JoinMode::Semi);
+  bool inputs_swapped = (_mode == JoinMode::Left || _mode == JoinMode::AntiDiscardNulls ||
+                         _mode == JoinMode::AntiRetainNulls || _mode == JoinMode::Semi);
 
   // (3) else the smaller relation will become build relation, the larger probe relation
-  if (!inputs_swapped && _input_left->get_output()->row_count() > _input_right->get_output()->row_count()) {
+  // (4) in case of the right outer join the right table must always remain the probe relation
```

Comment:
In the Github diff, it looks like there is nothing under (3), is that correct?

Reply:
That's correct. Also there is nothing under (1) and (2) now. The comment is more like a listing of cases to explain when we have to swap.